annotations/views/relationtemplate_views.py

```diff
@@ -201,15 +202,76 @@ def create_from_relationtemplate(request, template_id):
         text = get_object_or_404(Text, pk=data['occursIn'])
         project_id = data.get('project')
         if project_id is None:
-            project_id = VogonUserDefaultProject.objects.get(for_user=request.user).project.id
-        relationset = relations.create_relationset(template, data, request.user, text, project_id)
+            project_id = VogonUserDefaultProject.objects.get(
+                for_user=request.user).project.id
+        relationset = relations.create_relationset(
+            template, data, request.user, text, project_id)
         response_data = {'relationset': relationset.id}
-    else:   # Not sure if we want to do anything for GET requests at this point.
+    else:  # Not sure if we want to do anything for GET requests at this point.
         response_data = {}
 
     return JsonResponse(response_data)
 
 
+def create_from_text(request, template_id):
+    if request.method == 'POST':
+        data = json.loads(request.body)
+        appellations = data['appellations']
+        text_appellation = data['textAppellation']
+        template = get_object_or_404(RelationTemplate, pk=template_id)
+        text = get_object_or_404(Text, pk=data['occursIn'])
+        project_id = data.get('project')
+        if project_id is None:
+            project_id = VogonUserDefaultProject.objects.get(
+                for_user=request.user).project.id
+        for appellation in appellations:
+            #FIXME: Hardcoding this data object is not good practice
+            # but will work for the time being
+            appellation_object = {
+                'end':
+                None,
+                'fields': [{
+                    'appellation': text_appellation,
+                    'part_field': 'source',
+                    'description': '',
+                    'concept_label': None,
+                    'evidence_reqired': False,
+                    'label': 'Text',
+                    'part_id': data['part_id'],
+                    'type': 'TP',
+                    'concept_id': None
+                },
+                           {
+                               'appellation': appellation,
+                               'part_field': 'object',
+                               'description': '',
+                               'concept_label': None,
+                               'evidence_reqired': True,
+                               'label': 'Concept',
+                               'part_id': data['part_id'],
+                               'type': 'TP',
+                               'concept_id': None
+                           }],
```

Comment:
indentation seems off here?

Reply:
This is the work of the good old formatter, I will change to something not so strange.